custom_connectors/custom_auth/cornerstone_ondemand.rb

```diff
@@ -0,0 +1,137 @@
+{
+  title: 'Cornerstone OnDemand',
+
+  connection: {
+    fields: [
+      {
+        name: 'corpname',
+        optional: false
+      },
+      {
+        name: 'api_key',
+        optional: false
+      },
+      {
+        name: 'api_secret',
+        optional: false,
+        control_type: 'password'
+      },
+      {
+        name: 'user_name',
+        optional: false
+      },
+      {
+        name: 'alias',
+        optional: false
+      }
+    ],
+
+    # We use a custom authorization scheme:
+    # https://docs.workato.com/developing-connectors/sdk/authentication/custom-authentication.html
+    authorization: {
+      # How to acquire the token:
+      # https://docs.workato.com/developing-connectors/sdk/authentication/custom-authentication.html#acquire
+      acquire: lambda do |connection|
+        request = call(
+          :signed_request,
+          connection['api_secret'],
+          'POST',
+          connection['corpname'],
+          '/services/api/sts/session',
+          { 'x-csod-api-key': connection['api_key'] },
+          { userName: connection['user_name'], alias: connection['alias'] },
+          ''
+        )
+        
+        api_token = request.
+          format_xml(nil, strip_response_namespaces: true).
+          # Request lazy-executed here.
+          # XPath-like walk of the API response
+          dig('cornerstoneApi', 0, 'data', 0, 'Session', 0, 'Token', 0, 'content!')
+
+        # Result is added to the connection hash
+        { api_token: api_token }
+      end,
+
+      # How to know when to re-acquire the token.  Short-hand version of:
+      # https://docs.workato.com/developing-connectors/sdk/authentication/custom-authentication.html#refreshon
+      refresh_on: 401,
+
+      apply: lambda do |connection|
+        path = current_url.gsub("https://#{connection['corpname'].csod.com", '').gsub(/\?.*$/, '')
+        timestamp = Time.now.utc.strftime('%Y-%m-%dT%H:%M:%S.%3N')
+        msg = [
```

Comment:
Why not call the sign method here?

Reply:
Argument list is different in the two places it would be invoked, so it wouldn't actually help much in terms of DRY.